materialize/resources/resource_connection_aws_privatelink.go

```diff
@@ -0,0 +1,209 @@
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "public",
+	},
+	"database_name": {
+		Description: "The identifier for the connection database.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "materialize",
+	},
+	"qualified_name": {
+		Description: "The fully qualified name of the connection.",
+		Type:        schema.TypeString,
+		Computed:    true,
+	},
+	"service_name": {
+		Description:  "The name of the AWS PrivateLink service.",
+		Type:         schema.TypeString,
+		Optional:     true,
+		ForceNew:     true,
+		RequiredWith: []string{"availability_zones"},
+	},
+	"availability_zones": {
+		Description:  "The availability zones of the AWS PrivateLink service.",
+		Type:         schema.TypeList,
+		Elem:         &schema.Schema{Type: schema.TypeString},
+		Optional:     true,
+		ForceNew:     true,
+		RequiredWith: []string{"service_name"},
+	},
+}
+
+func ConnectionAwsPrivatelink() *schema.Resource {
+	return &schema.Resource{
+		Description: "The connection resource allows you to manage connections in Materialize.",
+
+		CreateContext: connectionAwsPrivatelinkCreate,
+		ReadContext:   connectionAwsPrivatelinkRead,
+		UpdateContext: connectionAwsPrivatelinkUpdate,
+		DeleteContext: connectionAwsPrivatelinkDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: connectionAwsPrivatelinkSchema,
+	}
+}
+
+type ConnectionAwsPrivatelinkBuilder struct {
+	connectionName               string
+	schemaName                   string
+	databaseName                 string
+	connectionType               string
+	privateLinkServiceName       string
+	privateLinkAvailabilityZones []string
+}
+
+func newConnectionAwsPrivatelinkBuilder(connectionName, schemaName, databaseName string) *ConnectionAwsPrivatelinkBuilder {
+	return &ConnectionAwsPrivatelinkBuilder{
+		connectionName: connectionName,
+		schemaName:     schemaName,
+		databaseName:   databaseName,
+	}
+}
+
+func (b *ConnectionAwsPrivatelinkBuilder) ConnectionName(connectionName string) *ConnectionAwsPrivatelinkBuilder {
+	b.connectionName = connectionName
+	return b
+}
+
+func (b *ConnectionAwsPrivatelinkBuilder) SchemaName(schemaName string) *ConnectionAwsPrivatelinkBuilder {
+	b.schemaName = schemaName
+	return b
+}
+
+func (b *ConnectionAwsPrivatelinkBuilder) PrivateLinkServiceName(privateLinkServiceName string) *ConnectionAwsPrivatelinkBuilder {
+	b.privateLinkServiceName = privateLinkServiceName
+	return b
+}
+
+func (b *ConnectionAwsPrivatelinkBuilder) PrivateLinkAvailabilityZones(privateLinkAvailabilityZones []string) *ConnectionAwsPrivatelinkBuilder {
+	b.privateLinkAvailabilityZones = privateLinkAvailabilityZones
+	return b
+}
+
+func (b *ConnectionAwsPrivatelinkBuilder) Create() string {
+	q := strings.Builder{}
+	q.WriteString(fmt.Sprintf(`CREATE CONNECTION %s.%s.%s TO AWS PRIVATELINK (`, b.databaseName, b.schemaName, b.connectionName))
```

Comment:
Out of scope for this PR but maybe we should make `qualified_name` a func on the builder type. Would be cleaner to just have that than the database, schema, name string parsing in the Create, Drop and Rename SQL generators.